Database: Upgrade dependency packages (24.11) [STUD-69625]

viogroza opened the pull request:
Upgrade Microsoft.Data.SqlClient to 5.2.1
Upgrade Oracle.ManagedDataAccess to 19.22.0
Upgrade Oracle.ManagedDataAccess.Core to 23.5.0
Fix custom providers for legacy
Upgrade Microsoft.Data.SqlClient.SNI.dll and Microsoft.Data.SqlClient.SNI.x86.dll 

I upgraded only the above packages to the latest version that was still compatible with Net461 (where applicable)

https://uipath.atlassian.net/browse/STUD-69625

sorinconstantin commented on Activities/Database/UiPath.Database/UiPath.Database.csproj:
```diff
@@ -19,18 +19,18 @@
     <PackageReference Include="System.Data.OleDb" Version="7.0.0" />
     <PackageReference Include="UiPath.Workflow" />
-    <PackageReference Include="Oracle.ManagedDataAccess.Core" Version="3.21.90" PrivateAssets="All" />
+    <PackageReference Include="Oracle.ManagedDataAccess.Core" Version="23.5.0" PrivateAssets="All" />
```
now that's a huge leap forward 🚀 
I hope it doesn't add any breaking changes

viogroza replied:
they changed the versioning system.. it jumps from 3.xx to 23.xx
https://www.nuget.org/packages/Oracle.ManagedDataAccess.Core/#versions-body-tab

viogroza committed:
Database: Upgrade packages (24.11) [STUD-69625]

Upgrade Microsoft.Data.SqlClient to 5.2.1
Upgrade Oracle.ManagedDataAccess to 19.22.0
Upgrade Oracle.ManagedDataAccess.Core to 23.5.0
Fix custom providers for legacy
Upgrade Microsoft.Data.SqlClient.SNI.dll and Microsoft.Data.SqlClient.SNI.x86.dll

alexandru-petre commented on Activities/Database/UiPath.Database.Activities.Design/UiPath.Database.Activities.Design.csproj:
```diff
@@ -11,7 +11,7 @@
     <PackageReference Include="UiPath.Workflow" />
     <PackageReference Include="System.Activities.Core.Presentation" />
-	<PackageReference Include="Oracle.ManagedDataAccess.Core" Version="3.21.90" PrivateAssets="All" />
+	<PackageReference Include="Oracle.ManagedDataAccess.Core" Version="23.5.0" PrivateAssets="All" />
```
nit: TAB / space misalignment
